File: 2165-smallest-value-of-the-rearranged-number/2165-smallest-value-of-the-rearranged-number.java
class Solution {
    public long smallestNumber(long num) {
       String s=String.valueOf(num);
    char[] ans=s.toCharArray();
     Arrays.sort(ans);
    if(num<0){
     String ss="-";
        for(int i=ans.length-1;i>=1;i--){
            ss+=ans[i];
        }
        Long ln=Long.parseLong(ss);  
        return ln;
         }  
    else{
        String sss="";
    if(ans[0]=='0'){
        int k=1;
        while(k<ans.length && ans[k]=='0'){
            k++;
        }
        if(k==ans.length)return 0;
        swap(ans,k);
    }
      for(int i=0;i<ans.length;i++){
            sss+=ans[i];
        }
     Long ln=Long.parseLong(sss);  
        return ln;
         
    }
}
public void swap(char [] ans,int k){
    char ch=ans[k];
    ans[k]=ans[0];
    ans[0]=ch;
}
 
    
}

File: 2165-smallest-value-of-the-rearranged-number/2165-smallest-value-of-the-rearranged-number.cpp
class Solution {
public:
    long long smallestNumber(long long num) {
       string s = to_string(num);
   if (num > 0) {
   	sort(s.begin(), s.end());
   	if (s[0] == '0') {
   		int i = 0;
   		while (s[i] == '0') {
   			i++;
   		}
   		swap(s[0], s[i]);
   	}
   }
   else if (num == 0)
   	return num;
   else {
   	sort(s.begin() + 1, s.end(), greater<char>());
   }
   return stoll(s); 
    }
};